pandas/core/generic.py

```diff
@@ -2155,6 +2155,7 @@ def to_excel(
         inf_rep: str = "inf",
         freeze_panes: tuple[int, int] | None = None,
         storage_options: StorageOptions = None,
+        engine_kwargs: dict | None = None,
```

Comment:
Can type-hint as `dict[str, Any] | None` I think.

Reply:
updating to type-hint `dict[str, Any] | None = None` as the above causes a syntax error. Would you consider this to be acceptable?

![image](https://github.com/pandas-dev/pandas/assets/45905457/8d29248e-4ecf-4fba-9766-c24289fe8ab9)

Reply:
Yes - you need to keep the `= None`; the request was for the type-hint portion.